app/controllers/images_controller.rb

```diff
@@ -12,28 +12,21 @@ def index
   end
 
   def create
-    Edition.find_and_lock_current(document: params[:document]) do |edition|
-      @issues = ::Requirements::ImageUploadChecker.new(params[:image]).issues
-
-      if @issues.any?
-        flash.now["alert_with_items"] = {
-          "title" => I18n.t!("images.index.flashes.upload_requirements"),
-          "items" => @issues.items,
-        }
-
-        render :index,
-               assigns: { edition: edition },
-               layout: rendering_context,
-               status: :unprocessable_entity
-        next
-      end
-
-      image_revision = ImageUploadService.new(params[:image], edition.revision).call(current_user)
-      updater = Versioning::RevisionUpdater.new(edition.revision, current_user)
-      updater.update_image(image_revision)
-      edition.assign_revision(updater.next_revision, current_user).save!
-      PreviewService.new(edition).try_create_preview
-      redirect_to crop_image_path(params[:document], image_revision.image_id, wizard: "upload")
+    result = Images::Create.call(params: params, user: current_user)
+    if result.failure?
```

Comment:
I also notice the examples in the gem use `result.success?` in preference to `failure?`. Probably doesn't make much difference, except that having the success code first communicates the intention of the action faster.

Reply:
Yeah I don't mind if success is before failure, I imagine I mostly maintained the existing flow of the code.

> What do you think this would look like for distinct types of failure?

Could do something like:

```
if result.failure_reason == :invalid_state
elsif result failure.reason == :issues
elsif result.success?
end
```
Though I'd hope we're mostly only dealing with one failure type in the view (for the sake of conditionals) and that other pre-condition ones can be handled with exceptions in a similar way to `ActiveRecord::RecordNotFound`.

Reply:
I'm not too fussed either - I was trying to think of a way of maintaining the clarity of the current controller structure when we have requirements issues. Perhaps just doing `raise unless result.issues` is enough to express that in `result.failure?`, as using `if`s can still imply there's a third execution path, which is neither a defined success or failure.